### PR TITLE
[API] Update VerifyCustomerAccount to the new metadata

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | api-platform-3 |
| Bug fix?        | no                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | -                      |
| License         | MIT                                                          |

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            >
+                <validationContext>
+                    <values>
+                        <value name="groups">
+                            <values>
+                                <value>sylius</value>
+                            </values>
+                        </value>
+                    </values>
+                </validationContext>
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml]: We can set this by default for the entire resource

File: src/Sylius/Bundle/ApiBundle/StateProcessor/Input/LoggedInShopUserIdAwareInputDataProcessor.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+/** @experimental */
+final readonly class LoggedInShopUserIdAwareInputDataProcessor implements InputDataProcessorInterface
+{
+    public function __construct(private UserContextInterface $userContext)
+    {
+    }
+
+    public function process(mixed $data, Operation $operation, array $uriVariables = [], array $context = []): mixed
+    {
+        Assert::isInstanceOf($data, ShopUserIdAwareInterface::class);
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/StateProcessor/Input/LoggedInShopUserIdAwareInputDataProcessor.php:31]: Is a double check needed?

**Review by Wojdylak — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/StateProcessor/Input/LoggedInShopUserIdAwareInputDataProcessor.php (hunk 2 above)
Comment (Wojdylak) [reply src/Sylius/Bundle/ApiBundle/StateProcessor/Input/LoggedInShopUserIdAwareInputDataProcessor.php:31]: This class will probably be removed. #15433

**Review by Wojdylak — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Customer.xml [hunk 3]
```diff
@@ -28,6 +28,29 @@
         </validationContext>
 
         <operations>
+            <operation
+                class="ApiPlatform\Metadata\Post"
+                uriTemplate="shop/customers"
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Customer.xml]:
```suggestion
                uriTemplate="/shop/customers"
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                class="ApiPlatform\Metadata\Patch"
+                uriTemplate="/account-verification-requests/{token}"
+                input="Sylius\Bundle\ApiBundle\Command\Account\VerifyCustomerAccount"
+                provider="Sylius\Bundle\ApiBundle\StateProvider\Input\VerifyCustomerAccountStateProvider"
+            >
+                <denormalizationContext>
+                    <values>
+                        <value name="groups">
+                            <values>
+                                <value>shop:account_verification:update</value>
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml]: Would be good to specify which account.

```suggestion
                                <value>shop:shop_user_account_verification:update</value>
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+ file that was distributed with this source code.
+
+-->
+
+<resources xmlns="https://api-platform.com/schema/metadata/resources-3.0"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="https://api-platform.com/schema/metadata/resources-3.0 https://api-platform.com/schema/metadata/resources-3.0.xsd"
+>
+    <resource
+        class="Sylius\Bundle\ApiBundle\Command\Account\VerifyCustomerAccount"
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml]: Overall, should naming be derived from `Customer` and not `ShopUser`? As internal implementation and not public facing endpoint, it should follow core terminology.

```suggestion
        class="Sylius\Bundle\ApiBundle\Command\Account\VerifyShopUser"
```

Comment (diimpp) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml]: Another pros for renaming, registration is done against ShopUser. 
`Sylius\Bundle\ApiBundle\Command\Account\RegisterShopUser`

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Customer.xml [hunk 6]
```diff
@@ -28,6 +28,29 @@
...
+                        <value name="groups">
+                            <values>
+                                <value>shop:customer:create</value>
+                            </values>
+                        </value>
+                    </values>
+                </denormalizationContext>
+                <openapiContext>
+                    <values>
+                        <value name="summary">Registers a new customer</value>
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Customer.xml]: I'm not sure about this ambiguity, sylius has strict `ShopUser`/`Customer` separation, new api is very explicit in terms how it exposes internals. e.g. if api has adjustments endpoint, then I see no reason to hide `ShopUser`.

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+                    <values>
+                        <value>sylius</value>
+                    </values>
+                </value>
+            </values>
+        </validationContext>
+
+        <operations>
+            <operation
+                name="shop_resend_verification_email"
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml]: Naming wise, this and command class should be changed to derivative of "Request verification token".

https://github.com/Sylius/Sylius/blob/a88c39a9de5a3650fc814de51f21bad37a0224d9/src/Sylius/Bundle/UserBundle/Controller/UserController.php#L160

```suggestion
                name="shop_request_verification"
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                class="ApiPlatform\Metadata\Post"
+                uriTemplate="/account-verification-requests"
+                input="Sylius\Bundle\ApiBundle\Command\Account\ResendVerificationEmail"
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_platform/resources/Command/VerifyCustomerAccount.xml]:
```suggestion
                input="Sylius\Bundle\ApiBundle\Command\Account\RequestShopUserVerification"
```
```suggestion
                input="Sylius\Bundle\ApiBundle\Command\Account\RequestShopUserVerificationToken"
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/command_handlers.xml [hunk 9]
```diff
@@ -158,35 +158,35 @@
...
 <!--        </service>-->
 
-<!--        <service id="Sylius\Bundle\ApiBundle\CommandHandler\Account\SendAccountRegistrationEmailHandler">-->
-<!--            <argument type="service" id="sylius.repository.shop_user" />-->
-<!--            <argument type="service" id="sylius.repository.channel" />-->
-<!--            <argument type="service" id="sylius.email_sender" />-->
-<!--            <tag name="messenger.message_handler" bus="sylius.command_bus" />-->
-<!--            <tag name="messenger.message_handler" bus="sylius_default.bus" />-->
-<!--        </service>-->
+        <service id="Sylius\Bundle\ApiBundle\CommandHandler\Account\SendAccountRegistrationEmailHandler">
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/command_handlers.xml:175]:
```suggestion
        <service id="Sylius\Bundle\ApiBundle\CommandHandler\Account\SendShopUserRegistrationEmailHandler">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/command_handlers.xml [hunk 10]
```diff
@@ -158,35 +158,35 @@
...
+        </service>
 
-<!--        <service id="Sylius\Bundle\ApiBundle\CommandHandler\Account\SendAccountVerificationEmailHandler">-->
-<!--            <argument type="service" id="sylius.repository.shop_user" />-->
-<!--            <argument type="service" id="sylius.repository.channel" />-->
-<!--            <argument type="service" id="sylius.email_sender" />-->
-<!--            <tag name="messenger.message_handler" bus="sylius.command_bus" />-->
-<!--            <tag name="messenger.message_handler" bus="sylius_default.bus" />-->
-<!--        </service>-->
+        <service id="Sylius\Bundle\ApiBundle\CommandHandler\Account\SendAccountVerificationEmailHandler">
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/command_handlers.xml]:
```suggestion
        <service id="Sylius\Bundle\ApiBundle\CommandHandler\Account\RequestShopUserVerificationHandler">
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/command_handlers.xml [hunk 11]
```diff
@@ -210,13 +210,13 @@
...
 <!--        </service>-->
 
-<!--        <service id="Sylius\Bundle\ApiBundle\CommandHandler\Account\VerifyCustomerAccountHandler">-->
-<!--            <argument type="service" id="sylius.repository.shop_user" />-->
-<!--            <argument type="service" id="clock" />-->
-<!--            <argument type="service" id="sylius.command_bus" />-->
-<!--            <tag name="messenger.message_handler" bus="sylius.command_bus" />-->
-<!--            <tag name="messenger.message_handler" bus="sylius_default.bus" />-->
-<!--        </service>-->
+        <service id="Sylius\Bundle\ApiBundle\CommandHandler\Account\VerifyCustomerAccountHandler">
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/command_handlers.xml]:
```suggestion
        <service id="Sylius\Bundle\ApiBundle\CommandHandler\Account\VerifyShopUserHandler">
```

**Review by diimpp — COMMENTED** [review]
Let's synchronize ShopUser/Customer terminology where it's needed.

**Review by Wojdylak — APPROVED** [review]

**Review by jakubtobiasz — APPROVED** [review]

**Pull request merged**